docs/extend: Mention and link to CLI configuration section on Development Overrides for Provider Developers (#1586)
hashicorp/terraform-website: content/source/docs/extend/testing/index.html.md

deliver on our mission to safely and predictably create, change, and improve
infrastructure. Documentation for Terraform tests are broken into categories
briefly described below. Each category has more detailed information by clicking
on the matching item in the left navigation.

> -> Recent versions of Terraform CLI also support [developer overrides in the CLI configuration](https://www.terraform.io/docs/commands/cli-config.html#development-overrides-for-provider-developers), which may be useful for manually testing providers. The acceptance testing framework uses real Terraform CLI executions, so developer overrides is only recommended as a last resort option for missing functionality.

Reviewer: ```suggestion
-> **Note:** Recent versions of Terraform CLI also support [developer overrides in the CLI configuration](/docs/commands/cli-config.html#development-overrides-for-provider-developers), which can be useful for manually testing providers. The acceptance testing framework uses real Terraform CLI executions, so developer overrides are only recommended as a last resort option for missing functionality.
```
